bindings/matrix-sdk-ffi/src/room.rs

```diff
@@ -636,6 +645,151 @@ impl Room {
             Ok(())
         })
     }
+
+    pub fn send_image(
+        &self,
+        url: String,
+        thumbnail_url: String,
+        image_info: ImageInfo,
+    ) -> Result<(), RoomError> {
+        let mime_type = match &image_info.mimetype {
+            Some(mimetype) => {
+                mimetype.parse::<Mime>().map_err(|_| RoomError::InvalidAttachmentMimeType)
+            }
+            None => Err(RoomError::InvalidAttachmentMimeType),
+        }?;
```

Comment:
```suggestion
        let mime_str = image_info.mimetype
            .as_ref()
            .ok_or(RoomError::InvalidAttachmentMimeType)?;
        let mime_type =
            mime_str.parse::<Mime>().map_err(|_| RoomError::InvalidAttachmentMimeType)?;
```

Though why does `image_info.mimetype == None` result in an `InvalidAttachmentMimeType` error? "Not set" and "invalid" seem like two different things to me.

Reply:
Why is yours better than the other version? 🤔 

> Though why does image_info.mimetype == None result in an InvalidAttachmentMimeType error?

Because I wanted to reuse the existing `ImageInfo` struct that has all those properties as optionals, presumably so that it can deal with historical data and other clients. 
I don't believe we should continue allowing that going forward so this enforces it at runtime.

Reply:
> Why is yours better than the other version? 🤔

Less nesting, error propagation happens exactly where the error is created rather than a few lines below; also, `?` on a `match` if `if` expression looks weird is sometimes easy to miss.

> > Though why does image_info.mimetype == None result in an InvalidAttachmentMimeType error?
> 
> Because I wanted to reuse the existing `ImageInfo` struct that has all those properties as optionals, presumably so that it can deal with historical data and other clients. I don't believe we should continue allowing that going forward so this enforces it at runtime.

Okay, we can merge it like this and possibly revisit later.

Reply:
> Less nesting, error propagation happens exactly where the error is created rather than a few lines below; also, ? on a match if if expression looks weird is sometimes easy to miss.

Fair enough, makes total sense, thanks for explaining 👍